### ISSUE TITLE
Break after the first epoch

### ISSUE THREAD

**Issue by jevrub** [ultralytics/yolov5#11192]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I start training and looks like it breaks after the first epoch. Where I should start looking at? 

### Additional

.yaml : 

train: ../dataset/train/images # labels are under ../dataset/train/labels in format className CX, CY, W, H (normalized)
val: ../dataset/val/images # labels are under ../dataset/val/labels in format className CX, CY, W, H (normalized)

nc: 20

names: [
    "road",
    "sidewalk",
    "building",
    "wall",
    "fence",
    "pole",
    "traffic light",
    "traffic sign",
    "vegetation",
    "terrain" ,
    "sky" ,
    "person" ,
    "rider" ,
    "car" ,
    "truck" ,
    "bus" ,
    "train" ,
    "motorcycle" ,
    "bicycle" ,
    "license plate"
  ]

Calling:
!python train.py --data ../cityscapes.yaml --weights yolov5m.pt --img 640 --epochs 10 --batch-size 16 --cache ram --cache disk

Output:
train: weights=yolov5m.pt, cfg=, data=../cityscapes.yaml, hyp=data/hyps/hyp.scratch-low.yaml, epochs=10, batch_size=16, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, noplots=False, evolve=None, bucket=, cache=disk, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[0], save_period=-1, seed=0, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
github: up to date with https://github.com/ultralytics/yolov5 ✅
YOLOv5 🚀 v7.0-120-g3e55763 Python-3.9.16 torch-1.13.1+cu116 CPU

hyperparameters: lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0
ClearML: run 'pip install clearml' to automatically track, visualize and remotely train YOLOv5 🚀 in ClearML
Comet: run 'pip install comet_ml' to automatically track and visualize YOLOv5 🚀 runs in Comet
TensorBoard: Start with 'tensorboard --logdir runs/train', view at http://localhost:6006/
2023-03-18 14:29:50.448712: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-18 14:29:53.223877: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.9/dist-packages/cv2/../../lib64:/usr/local/lib/python3.9/dist-packages/cv2/../../lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2023-03-18 14:29:53.224054: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.9/dist-packages/cv2/../../lib64:/usr/local/lib/python3.9/dist-packages/cv2/../../lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2023-03-18 14:29:53.224078: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
Overriding model.yaml nc=80 with nc=20

                 from  n    params  module                                  arguments                     
  0                -1  1      5280  models.common.Conv                      [3, 48, 6, 2, 2]              
  1                -1  1     41664  models.common.Conv                      [48, 96, 3, 2]                
  2                -1  2     65280  models.common.C3                        [96, 96, 2]                   
  3                -1  1    166272  models.common.Conv                      [96, 192, 3, 2]               
  4                -1  4    444672  models.common.C3                        [192, 192, 4]                 
  5                -1  1    664320  models.common.Conv                      [192, 384, 3, 2]              
  6                -1  6   2512896  models.common.C3                        [384, 384, 6]                 
  7                -1  1   2655744  models.common.Conv                      [384, 768, 3, 2]              
  8                -1  2   4134912  models.common.C3                        [768, 768, 2]                 
  9                -1  1   1476864  models.common.SPPF                      [768, 768, 5]                 
 10                -1  1    295680  models.common.Conv                      [768, 384, 1, 1]              
 11                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 12           [-1, 6]  1         0  models.common.Concat                    [1]                           
 13                -1  2   1182720  models.common.C3                        [768, 384, 2, False]          
 14                -1  1     74112  models.common.Conv                      [384, 192, 1, 1]              
 15                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 16           [-1, 4]  1         0  models.common.Concat                    [1]                           
 17                -1  2    296448  models.common.C3                        [384, 192, 2, False]          
 18                -1  1    332160  models.common.Conv                      [192, 192, 3, 2]              
 19          [-1, 14]  1         0  models.common.Concat                    [1]                           
 20                -1  2   1035264  models.common.C3                        [384, 384, 2, False]          
 21                -1  1   1327872  models.common.Conv                      [384, 384, 3, 2]              
 22          [-1, 10]  1         0  models.common.Concat                    [1]                           
 23                -1  2   4134912  models.common.C3                        [768, 768, 2, False]          
 24      [17, 20, 23]  1    101025  models.yolo.Detect                      [20, [[10, 13, 16, 30, 33, 23], [30, 61, 62, 45, 59, 119], [116, 90, 156, 198, 373, 326]], [192, 384, 768]]
Model summary: 291 layers, 20948097 parameters, 20948097 gradients, 48.5 GFLOPs

Transferred 475/481 items from yolov5m.pt
optimizer: SGD(lr=0.01) with parameter groups 79 weight(decay=0.0), 82 weight(decay=0.0005), 82 bias
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
train: Scanning /content/dataset/train/labels.cache... 174 images, 0 backgrounds, 0 corrupt: 100% 174/174 [00:00<?, ?it/s]
train: WARNING ⚠️ /content/dataset/train/images/aachen_000000_000019_gtFine_color.png: 6 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000002_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000004_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000008_000019_gtFine_color.png: 2 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000022_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000026_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000030_000019_gtFine_color.png: 4 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000031_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000034_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000042_000019_gtFine_color.png: 3 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000054_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000058_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000060_000019_gtFine_color.png: 24 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000061_000019_gtFine_color.png: 41 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000062_000019_gtFine_color.png: 9 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000064_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000070_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000072_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000078_000019_gtFine_color.png: 4 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000079_000019_gtFine_color.png: 5 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000080_000019_gtFine_color.png: 45 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000090_000019_gtFine_color.png: 2 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000092_000019_gtFine_color.png: 1 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000095_000019_gtFine_color.png: 2 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000096_000019_gtFine_color.png: 3 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000098_000019_gtFine_color.png: 2 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000152_000019_gtFine_color.png: 28 duplicate labels removed
train: WARNING ⚠️ /content/dataset/train/images/aachen_000155_000019_gtFine_color.png: 1 duplicate labels removed
train: Caching images (1.0GB disk): 100% 174/174 [00:08<00:00, 20.38it/s]
val: Scanning /content/dataset/val/labels.cache... 267 images, 0 backgrounds, 0 corrupt: 100% 267/267 [00:00<?, ?it/s]
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_016273_gtFine_color.png: 1 duplicate labels removed
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_017101_gtFine_color.png: 1 duplicate labels removed
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_028232_gtFine_color.png: 1 duplicate labels removed
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_044787_gtFine_color.png: 1 duplicate labels removed
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_046272_gtFine_color.png: 1 duplicate labels removed
val: WARNING ⚠️ /content/dataset/val/images/frankfurt_000001_077434_gtFine_color.png: 1 duplicate labels removed
val: Caching images (1.6GB disk): 100% 267/267 [00:15<00:00, 17.41it/s]

AutoAnchor: 2.65 anchors/target, 0.812 Best Possible Recall (BPR). Anchors are a poor fit to dataset ⚠️, attempting to improve...
AutoAnchor: WARNING ⚠️ Extremely small objects found: 1931 of 10721 labels are <3 pixels in size
AutoAnchor: Running kmeans for 9 anchors on 10708 points...
AutoAnchor: Evolving anchors with Genetic Algorithm: fitness = 0.6052: 100% 1000/1000 [00:02<00:00, 356.95it/s]
AutoAnchor: thr=0.25: 0.9873 best possible recall, 2.87 anchors past thr
AutoAnchor: n=9, img_size=640, metric_all=0.207/0.607-mean/best, past_thr=0.450-mean: 5,4, 2,18, 10,10, 24,17, 11,41, 42,56, 148,31, 173,124, 515,162
AutoAnchor: Done ✅ (optional: update model *.yaml to use these anchors in the future)
Plotting labels to runs/train/exp13/labels.jpg... 
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to runs/train/exp13
Starting training for 10 epochs...

      Epoch    GPU_mem   box_loss   obj_loss   cls_loss  Instances       Size
        0/9         0G     0.1341     0.1236    0.08029       1637        640:   9% 1/11 [01:04<10:40, 64.01s/it]^C

P.S. I am not breaking it accidentally.

**Comment by glenn-jocher** [comment]
@jevrub it seems like the training process is being interrupted after the first epoch. The warning "AutoAnchor: Anchors are a poor fit to the dataset" indicates that the anchor boxes may not be suitable for your dataset, and "Extremely small objects found" suggests that some of your labels might be very small.

Here are a few things to consider:

- Double-check your dataset for extremely small objects and consider removing or resizing them.
- You may want to customize the anchor box sizes. You can do this by updating your .yaml file to use the anchors provided by the AutoAnchor tool during training.
- Additionally, consider altering the hyperparameters in your .yaml file, such as the box, cls, and obj settings.

For further details about configuring your dataset and modifying training parameters, refer to the YOLOv5 documentation at https://docs.ultralytics.com/yolov5/.

Let us know if you need further assistance!
